pkg/scanners/trivy/trivy.go

```diff
@@ -0,0 +1,397 @@
+
+	imgChan := make(chan string)
+	var wg sync.WaitGroup
+
+	for _, imageName := range scanList {
+		wg.Add(1)
+
+		go func(imageRef string) {
+			fmt.Printf("scanning: %s\n", imageRef)
+			dockerImage, cleanup, err := fanalImage.NewDockerImage(ctx, imageRef, scanConfig.dockerOptions)
+			if err != nil {
+				fmt.Fprintln(os.Stderr, err)
+				os.Exit(generalErr)
+			}
+			defer cleanup()
+
+			artifactToScan, err := artifactImage.NewArtifact(dockerImage, scanConfig.fscache, artifact.Option{}, config.ScannerOption{})
+			if err != nil {
+				fmt.Fprintln(os.Stderr, err)
+				os.Exit(generalErr)
+			}
+
+			scanner := scanner.NewScanner(scanConfig.localScanner, artifactToScan)
+
+			report, err := scanner.ScanArtifact(ctx, scanConfig.scanOptions)
+			if err != nil {
+				fmt.Fprintln(os.Stderr, err)
+				os.Exit(generalErr)
+			}
+
+		outer:
+			for i := range report.Results {
+				resultClient.FillVulnerabilityInfo(report.Results[i].Vulnerabilities, report.Results[i].Type)
+
+				for j := range report.Results[i].Vulnerabilities {
+					if *ignoreUnfixed && report.Results[i].Vulnerabilities[j].FixedVersion == "" {
+						continue
+					}
+
+					if report.Results[i].Vulnerabilities[j].Severity == "" {
+						report.Results[i].Vulnerabilities[j].Severity = severityUnknown
+					}
+
+					if severityMap[report.Results[i].Vulnerabilities[j].Severity] {
+						imgChan <- report.ArtifactName
+						break outer
+					}
+				}
+			}
+
+			wg.Done()
+		}(imageName)
+	}
+
+	go func() {
+		wg.Wait()
+		close(imgChan)
+	}()
+
+	vulnerableImages := make([]eraserv1alpha1.Image, 0, len(scanList))
+	for imageRef := range imgChan {
+		image := eraserv1alpha1.Image{Digest: "abc123", Name: imageRef}
+		vulnerableImages = append(vulnerableImages, image)
+	}
+
+	err = updateStatus(&statusUpdate{
+		apiPath:         apiPath,
+		ctx:             ctx,
+		clientset:       clientset,
+		collectorCRName: *collectorCRName,
+		resourceName:    resourceName,
+		subResourceName: subResourceName,
+		images:          vulnerableImages,
+	})
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(generalErr)
+	}
+
+	for _, imageRef := range vulnerableImages {
+		fmt.Println(imageRef)
+	}
+}
+
+// side effects: map `m` will be modified according to the values in `commaSeparatedList`.
+func parseCommaSeparatedOptions(m map[string]bool, commaSeparatedList string) error {
+	list := strings.Split(commaSeparatedList, ",")
+	for _, item := range list {
+		if _, ok := m[item]; !ok {
+			keys := mapKeys(m)
+			return fmt.Errorf("'%s' was not one of %#v", item, keys)
+		}
+
+		m[item] = true
+	}
+
+	return nil
+}
+
+func downloadAndInitDB(cacheDir string) error {
+	err := downloadDB(cacheDir)
+	if err != nil {
+		return err
+	}
+
+	err = db.Init(cacheDir)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func downloadDB(cacheDir string) error {
+	client := dlDb.NewClient(cacheDir, true)
+	ctx := context.Background()
+	needsUpdate, err := client.NeedsUpdate("dev", false)
```

Comment:
what's `dev` here?

Reply:
It refers to the trivy version. This made its way in from looking at Trivy's state when it was running. I've corrected this now.